### PR TITLE
Bug Fix: Later js's weekday syntax is different from Quartz. 

### PR DESCRIPTION
**Pull request by fsi206914**
Unix Cron use 0-6 as Sun-Sat, but Quartz use 1-7. Due to later.js only supporting Unix Cron, we have to make this transition.

This change transform 1-7 to 0-6 in Cron's weekday field.

Tests are verified in our experimental cluster.

### CODE REVIEW

File: azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js [hunk 1]
```diff
@@ -186,12 +186,23 @@ var cron_output_id  = "#cron-output";
...
+// The input is a 5 field string (without year) or 6 field String (with year).
+function transformFromQuartzToUnixCron(str){
   var res = str.split(" ");
-  res[res.length -1] = res[res.length -1].replace(/[0-7]/g, function upperToHyphenLower(match) {
-    return (parseInt(match)+6)%7;
-  });
+
+  // If the cron doesn't include year field
+  if(res.length == 5)
+    res[res.length -1] = res[res.length -1].replace(/[0-7]/g, function upperToHyphenLower(match) {
```
Comment (HappyRay) [inline azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js]: Can you think of ways to reduce the logic duplication?

File: azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js [hunk 2]
```diff
@@ -186,12 +186,23 @@ var cron_output_id  = "#cron-output";
 var cron_translate_id  = "#cronTranslate";
 var cron_translate_warning_id  = "#translationWarning";
 
-// Cron use 0-6 as Sun--Sat, but Quartz use 1-7. Therefore, a translation is necessary.
-function transformFromCronToQuartz(str){
+// Uni Cron use 0-6 as Sun--Sat, but Quartz use 1-7. Due to later.js only supporting Unix Cron, we have to make this transition.
+// The detailed Unix Cron Syntax: https://en.wikipedia.org/wiki/Cron
+// The input is a 5 field string (without year) or 6 field String (with year).
+function transformFromQuartzToUnixCron(str){
```
Comment (HappyRay) [inline azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js]: Unit tests?

File: azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js (hunk 1 above)
Comment (fsi206914) [inline azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js]: Fixed by introducing a shared function.

File: azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js (hunk 2 above)
Comment (fsi206914) [inline azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js]: Will add unit test after importing mocha + chai.

**Comment by fsi206914** [comment]
I tried "update branch" button in this page. Do I still need to rebase myself? Let me know if it is good to merge.

**Comment by fsi206914** [comment]
Oops. Looks like I have to rebase locally due to suvodeep's change.

**Comment by hreview** [comment]
Hey Liang

Sorry about that. I had to push because my change was huge.
However, I don't think that you'll face any issues with it since they were
just file renames.

Regards
Suvodeep

**Comment by HappyRay** [comment]
Please resolve the conflict and feel free to merge.

In the future I would suggest keeping the unrelated changes separate e.g. the UI change seems to be independent from the syntax conversion.

**Comment by fsi206914** [comment]
Reopened this pull request in #738

**Pull request closed without merging**